firefox-ios/Client/Frontend/Library/Bookmarks/Utitlity/BookmarksSaver.swift

```diff
@@ -19,51 +19,70 @@ struct DefaultBookmarksSaver: BookmarksSaver {
 
     let profile: Profile
 
-    func save(bookmark: FxBookmarkNode, parentFolderGUID: String) async -> Result<Void, Error> {
+    func save(bookmark: FxBookmarkNode, parentFolderGUID: String) async -> Result<String?, Error> {
         return await withCheckedContinuation { continuation in
             let operation: Success? = {
                 switch bookmark.type {
                 case .bookmark:
                     guard let bookmark = bookmark as? BookmarkItemData else { return nil }
                     let position: UInt32? = parentFolderGUID == BookmarkRoots.MobileFolderGUID ? 0 : nil
-                    if bookmark.parentGUID == nil {
-                        return profile.places.createBookmark(parentGUID: parentFolderGUID,
-                                                             url: bookmark.url,
-                                                             title: bookmark.title,
-                                                             position: position).bind { result in
-                            return result.isFailure ? deferMaybe(BookmarkDetailPanelError()) : succeed()
+
+                    return bookmark.parentGUID == nil ?
```

Comment:
I'd restore this as before since i guess an if else is more readable here, and also i'd return the succeed() with the associated value and restore it like before in the final operation. So we call continuation in the final operation, that would be nice since we avoid referencing the continuation in many place since it could throw error if we call it more then one.